### ISSUE TITLE
issue with batchnorm mode 0 during exporting keras model to tensorflow

### ISSUE THREAD

**Issue by WenchenLi** [fchollet/keras#4118]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by WenchenLi** [comment]
hi 
I'm using keras 1.0.7 with tf backend to trained a  model successfully, now I want to export that for production. 

As pointed out by https://blog.keras.io/keras-as-a-simplified-interface-to-tensorflow-tutorial.html

I first rebuild the model 
`model = resnet.build_resnet50(1, 200, 200, max_nb_char, nb_classes)
`
then 
`s_preds, l_preds = model(x)`

I need to continue to train the model a bit with tf, so BatchNormalization mode was set to 0, 
but I got: 
```
File "/home/wenchen/tf_server/deep-captcha/tf_train.py", line 29, in <module>
    s_preds, l_preds = model(x)
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/topology.py", line 526, in __call__
    return self.call(x, mask)
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/topology.py", line 2017, in call
    output_tensors, output_masks, output_shapes = self.run_internal_graph(inputs, masks)
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/topology.py", line 2159, in run_internal_graph
    output_tensors = to_list(layer.call(computed_tensor, computed_mask))
  File "/usr/local/lib/python2.7/dist-packages/keras/layers/normalization.py", line 110, in call
    raise Exception('You are attempting to share a '
Exception: You are attempting to share a same `BatchNormalization` layer across different data flows. This is not possible. You should use `mode=2` in `BatchNormalization`, which has a similar behavior but is shareable (see docs for a description of the behavior).
```

I found a similar one https://github.com/fchollet/keras/issues/2954 but mine wasnt solved.
